client/state/themes/selectors.js

```diff
@@ -375,42 +392,22 @@ export function getThemeSignupUrl( state, theme ) {
 /**
  * Returns the currently active theme on a given site.
  *
- * DON'T USE YET! This relies on a site object's options.theme_slug attr. If you trigger my-sites' siteSelection
- * middleware during theme activation, it will fetch the current site fresh from the API even though that
- * theme_slug attr might not have been updated on the server yet -- and you'll end up with the old themeId!
- * This happens in particular after purchasing a premium theme in single-site mode since after a theme purchase,
- * the checkout-thank-you component always redirects to the theme showcase for the current site.
- * One possible fix would be to get rid of that redirect (related: https://github.com/Automattic/wp-calypso/issues/8262).
- *
```

Comment:
We should probably keep a comment explaining _why_ we duplicate information that's also available in `sites` state.